src/base/baseBrushTool.js

```diff
@@ -0,0 +1,367 @@
+import external from './../externalModules.js';
+import EVENTS from './../events.js';
+import baseTool from './../base/baseTool.js';
+// Utils
+import KeyboardController from '../fancy-tools/shared/KeyboardController.js';
+import isToolActive from '../fancy-tools/shared/isToolActive.js';
+import { getNewContext } from '../util/drawing.js';
+import { COLOR_MAP_ID } from '../stateManagement/brushToolColors.js';
+
+export default class extends baseTool {
+  constructor ({
+    name,
+    strategies,
+    defaultStrategy,
+    configuration,
+    supportedInteractionTypes
+  }) {
+    super({
+      name,
+      strategies,
+      defaultStrategy,
+      configuration,
+      supportedInteractionTypes
+    });
+
+    this.isBrushTool = true;
+    this.hasCursor = true;
+    this._referencedToolData = 'brush';
+
+    this._drawingMouseUpCallback = this._drawingMouseUpCallback.bind(this);
+  }
+
+  //===================================================================
+  // Abstract Methods - Must be implemented.
+  //===================================================================
+
+  /**
+  * Helper function for rendering the brush.
+  *
+  * @abstract
+  * @param {Object} evt - The event.
+  */
+  renderBrush (evt) {
+    throw new Error(`Method renderBrush not implemented for ${this.toolName}.`);
+  }
+
+  /**
+   * Paints the data to the canvas.
+   *
+   * @protected
+   * @abstract
+   * @param  {Object} eventData The data object associated with the event.
+   */
+  _paint (eventData) {
+    throw new Error(`Method renderBrush not implemented for ${this.toolName}.`);
+  }
+
+  //===================================================================
+  // Virtual Methods - Have default behavior but may be overriden.
+  //===================================================================
+
+  /**
+   * Callback for when the tool is activated.
+   *
+   * @virtual
+   */
+  activeCallback () {
+    const configuration = this.configuration;
+
+    this._changeDrawColor(configuration.draw);
```

Comment:
Can this be moved to the constructor? And then if you modify the configuration, or expose a method to change this property, it's applied when drawing?

Reply:
Yeah I think that'd make more sense, done.

The _changeDrawColor method is already wrapped and exposed by the keyboard interface. Do you intend to have a touch interface for the painting tools? We may need to expose it in another way if you wish to switch stuff via a menu on a phone. I feel like the UX of a brushTool interface may be tricky to get right on touch devices though.

Reply:
> Do you intend to have a touch interface for the painting tools?

Eventually? Yes. I could see some of our clinics using tablets or touch screen laptops like the Microsoft Surface. I'm not sure there's an _immediate_ need for Asteris.

Reply:
I reckon you could make it work, but the minimum brush size would have to be ~size of fingertip. This would be fine for quick drawing for demonstration etc. I'm assuming anyone who wants to draw out very detailed regions would use a computer?

Reply:
More than likely. The use case would be less diagnostic, and more demonstration. Our software is used in a few classrooms, and sometimes for patient education.